test/hotspot/jtreg/serviceability/dcmd/compiler/PerfMapTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test PerfMapTest
+ * @bug 8254723
+ * @requires os.family == "linux"
+ * @library /test/lib
+ * @modules java.base/jdk.internal.misc
+ *          java.compiler
+ *          java.management
+ *          jdk.internal.jvmstat/sun.jvmstat.monitor
+ * @run testng/othervm PerfMapTest
+ * @summary Test of diagnostic command Compiler.perfmap
+ */
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.dcmd.CommandExecutor;
+import jdk.test.lib.dcmd.JMXExecutor;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Iterator;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Call jcmd Compiler.perfmap and check the output file has the expected
+ * format.
+ */
+public class PerfMapTest {
+
+    static Pattern outputPattern =
+        Pattern.compile("Written to (/tmp/perf-\\p{Digit}*\\.map)");
+    static Pattern linePattern =
+        Pattern.compile("^((?:0x)?\\p{XDigit}+)\\s+((?:0x)?\\p{XDigit}+)\\s+(.*)$");
+
+    public void run(CommandExecutor executor) {
+        OutputAnalyzer output = executor.execute("Compiler.perfmap");
+        String line = output.asLines().iterator().next();
+        Matcher m = outputPattern.matcher(line);
+
+        Assert.assertTrue(m.matches(), "Did not print map file name");
+
+        // Sanity check the file contents
+        try (Scanner in = new Scanner(new FileInputStream(m.group(1)))) {
+            while (in.hasNextLine()) {
+                line = in.nextLine();
+                m = linePattern.matcher(line);
+                Assert.assertTrue(m.matches(), "Invalid file format: " + line);
+            }
+        } catch (FileNotFoundException e) {
+            Assert.fail(e.toString());
+        }
```

Comment:
Sounds like a lot of this scaffolding could be replaced by `for (String line : Files.readAllLines(m.group(1))) { ... }`?